inc/ctest_2_cppunittest.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef CTEST_2_CPPUNITTEST_H
+#define CTEST_2_CPPUNITTEST_H
+
+#include "macro_utils/macro_utils.h"
+
+// This structure matches 100% the structure used by CppUnitTest to store its metadata in the shared section where the metadata is stored
```

Comment:
>This structure matches 100% the structure used by CppUnitTest to store its metadata in the shared section where the metadata is stored [](start = 3, length = 134)

makes me wonder if there should be some version check somewhere (in cmakelists?) to gracefully fail in case this structure could change...maybe not worth the extra hassle, but maybe there should at least be some documentation here or in an MD about what version of the CppUnitTest code this is all based on.

Reply:
Added a compile time check for the size of the structures which would be a minimal check we can easily do

---
In reply to: [608143453](https://github.com/Azure/c-testrunnerswitcher/pull/65#discussion_r608143453) [](ancestors = 608143453)